src/Skia/Avalonia.Skia/GlyphTypefaceImpl.cs

```diff
@@ -13,10 +15,38 @@ internal class GlyphTypefaceImpl : IGlyphTypeface, IGlyphTypeface2
         private bool _isDisposed;
         private readonly SKTypeface _typeface;
 
+        private static uint GetIntTag(string v)
+        {
+            return (uint)v[0] << 24 | (uint)v[1] << 16 | (uint)v[2] << 08 | (uint)v[3] << 00;
+        }
+
+        public static bool TryReadFontMetrics(SKTypeface typeface, out int Ascent, out int Descent, out int LineGap)
+        {
+            // See: https://learn.microsoft.com/en-us/typography/opentype/spec/recom#baseline-to-baseline-distances
+            if (typeface.TryGetTableData(GetIntTag("OS/2"), out byte[]? os2Table) && 
+                typeface.TryGetTableData(GetIntTag("hhea"), out byte[]? hheaTable))
+            {
+                if (ReadOS2Table(os2Table, out int usWinAscent, out int usWinDescent) && 
```

Comment:
Do we need both tables? libgdiplus is using only one of those with os2 table taking priority:
https://github.com/mono/libgdiplus/blob/94a49875487e296376f209fe64b921c6020f74c0/src/font.c#L757-L792

Reply:
Hmm, actually, it analyzes the os2 table if it exists and uses its values instead of hhea if certain flags are set.

We probably want to just port that logic as is.

Reply:
I implemented the libgdiplus code in `GlyphTypefaceImpl` and this is what I got:

![image](https://github.com/AvaloniaUI/Avalonia/assets/32274404/c725c45e-7c23-4074-a7d1-8dba4703f4a8)

Maybe I implemented it incorrectly? I noticed that sTypoDescender was in the order of 65000 for fonts like Inter (which seems to be the first loaded font family), whereas usWinDescent was around 200.

This is the code I wrote:
```
private static bool TryReadFontMetrics(SKTypeface typeface, out int ascent, out int descent, out int lineGap)
{
    const int TagOs2 = 1330851634; // pre-computed value for HarfBuzzSharp.Tag.Parse("OS/2")
    const int TagHhea = 1751672161; // pre-computed value for HarfBuzzSharp.Tag.Parse("hhea")

    // See: https://learn.microsoft.com/en-us/typography/opentype/spec/recom#baseline-to-baseline-distances
    // See Also: https://github.com/mono/libgdiplus/blob/94a49875487e296376f209fe64b921c6020f74c0/src/font.c#L757-L792
    if (typeface.TryGetTableData(TagOs2, out byte[]? os2Table) && typeface.TryGetTableData(TagHhea, out byte[]? hheaTable))
    {
        if (ReadHHEATable(hheaTable, out int hheaAscender, out int hheaDescender, out int hheaLineGap))
        {
            bool hasOs2 = ReadOS2Table(os2Table, out OS2TableInfo os2);

            const int fsSelectionUseTypoMetrics = (1 << 7);
            if (hasOs2 && (os2.usWinAscent <= 0 || os2.usWinDescent <= 0))
            {
                // System.Diagnostics.Debugger.Break();
            }

            if (hasOs2 && (os2.fsSelection & fsSelectionUseTypoMetrics) != 0)
            {
                /* Use the typographic Ascender, Descender, and LineGap values for everything. */
                // This is the most common case using these values. sTypoDescender is huge, same with hheaDescender
                lineGap = os2.sTypoAscender - os2.sTypoDescender + os2.sTypoLineGap;
                descent = os2.sTypoDescender;
                ascent = os2.sTypoAscender;
            }
            else
            {
                /* Calculate the LineSpacing for both the hhea table and the OS/2 table. */
                int hhea_linespacing = hheaAscender + Math.Abs(hheaDescender) + hheaLineGap;
                int os2_linespacing = hasOs2 ? (os2.usWinAscent + os2.usWinDescent) : 0;

                /* The LineSpacing is the maximum of the two sumations. */
                lineGap = Math.Max(hhea_linespacing, os2_linespacing);

                /* If the OS/2 table exists, use usWinDescent as the
                 * CellDescent. Otherwise use hhea's Descender value. */
                descent = hasOs2 ? os2.usWinDescent : hheaDescender;

                /* If the OS/2 table exists, use usWinAscent as the
                 * CellAscent. Otherwise use hhea's Ascender value. */
                ascent = hasOs2 ? os2.usWinAscent : hheaAscender;
            }

            return true;
        }
    }

    ascent = descent = lineGap = 0;
    return false;
}

private static bool ReadOS2Table(byte[]? array, out OS2TableInfo os2)
{
    os2 = default;
    if (array == null || array.Length < 78)
    {
        return false;
    }

    // 62 is the offset to the fsSelection field in the OS/2 table
    var span = new Span<byte>(array, 62, 16);

    os2.fsSelection = BinaryPrimitives.ReadUInt16BigEndian(span);
    os2.sTypoAscender = BinaryPrimitives.ReadUInt16BigEndian(span.Slice(6));
    os2.sTypoDescender = BinaryPrimitives.ReadUInt16BigEndian(span.Slice(8));
    os2.sTypoLineGap = BinaryPrimitives.ReadUInt16BigEndian(span.Slice(10));
    os2.usWinAscent = BinaryPrimitives.ReadUInt16BigEndian(span.Slice(12));
    os2.usWinDescent = BinaryPrimitives.ReadUInt16BigEndian(span.Slice(14));
    return true;
}

private struct OS2TableInfo
{
    public ushort fsSelection;
    public ushort sTypoAscender;
    public ushort sTypoDescender;
    public ushort sTypoLineGap;
    public ushort usWinAscent;
    public ushort usWinDescent;
}
```